src/function/scalar/math/numeric.cpp

```diff
@@ -443,6 +443,89 @@ void FloorFun::RegisterFunction(BuiltinFunctions &set) {
 	set.AddFunction(floor);
 }
 
+//===--------------------------------------------------------------------===//
+// trunc
+//===--------------------------------------------------------------------===//
+struct TruncOperator {
+	// Integer truncation is a NOP
+	template <class TA, class TR>
+	static inline TR Operation(TA left) {
+		return left;
+	}
+};
+
+template <>
+inline float TruncOperator::Operation(float left) {
+	return std::trunc(left);
+}
+
+template <>
+inline double TruncOperator::Operation(double left) {
+	return std::trunc(left);
+}
+
+struct TruncDecimalOperator {
+	template <class T, class POWERS_OF_TEN_CLASS>
+	static void Operation(DataChunk &input, uint8_t scale, Vector &result) {
+		T power_of_ten = POWERS_OF_TEN_CLASS::POWERS_OF_TEN[scale];
+		UnaryExecutor::Execute<T, T>(input.data[0], result, input.size(), [&](T input) {
+			//	Always floor
+			return (input / power_of_ten);
+		});
+	}
+};
+
+void TruncFun::RegisterFunction(BuiltinFunctions &set) {
+	ScalarFunctionSet floor("trunc");
+	for (auto &type : LogicalType::Numeric()) {
+		scalar_function_t func = nullptr;
+		bind_scalar_function_t bind_func = nullptr;
+		//	Truncation of integers gets generated by some tools (e.g., Tableau/JDBC:Postgres)
+		switch (type.id()) {
+		case LogicalTypeId::FLOAT:
+			func = ScalarFunction::UnaryFunction<float, float, TruncOperator>;
+			break;
+		case LogicalTypeId::DOUBLE:
+			func = ScalarFunction::UnaryFunction<double, double, TruncOperator>;
+			break;
+		case LogicalTypeId::DECIMAL:
+			bind_func = BindGenericRoundFunctionDecimal<TruncDecimalOperator>;
+			break;
+		case LogicalTypeId::TINYINT:
+			func = ScalarFunction::UnaryFunction<int8_t, int8_t, TruncOperator>;
```

Comment:
This is generating a lot of code (and increasing binary footprint) for what are essentially nop-functions. Can we use `ScalarFunction::NopFunction` here instead?